src/components/ProductList/productList.js

```diff
@@ -0,0 +1,299 @@
+              Only fast Delivery
+            </label>
+          </div>
+        </div>
+      </div>
+
+      <div className="container__main">
+        <div className="card-container">
+          {!isLoading ? (
+            filteredData.map((data) => {
+              return (
+                <div className="card card--display" key={data._id}>
+                  <div className="card__thumbnail">
+                    <img
+                      src={data.imageUrl}
+                      className="card__img"
+                      alt="cardImg"
+                    />
+                  </div>
+                  <i
+                    className={
+                      isAddedInList(data._id, wishList)
+                        ? "fa fa-heart wish-icon wish-icon--selected"
+                        : "fa fa-heart wish-icon"
+                    }
+                    aria-hidden="true"
+                    onClick={() =>
+                      wishlistHandler(data, wishList, wishDispatch)
+                    }
+                  ></i>
+                  <div className="card__desc">
+                    <h1>
+                      <strong>{getTrimmedTitle(data.name)}</strong>
+                    </h1>
+                    <div className="star-count">
+                      <p className="star-count__star">{data.ratings}</p>
+                      <div className="rating">
+                        <div className="rating__stars">
+                          <i className="fa fa-star" aria-hidden="true"></i>
+                        </div>
+                      </div>
+                    </div>
+                    <h2>
+                      <strong> ₹ {data.price}</strong>
+                    </h2>
+                    <p
+                      className={
+                        data.inStock
+                          ? "stock-details inStock"
+                          : "stock-details outOfStock"
+                      }
+                    >
+                      {data.inStock ? "In Stock" : "Out of stock"}
+                    </p>
+                    <p className="card__details offer">{data.offer}</p>
+                    <button
+                      className={
+                        data.inStock
+                          ? "btn btn--primary btn--cart"
+                          : "btn btn--primary btn--cart disabled"
+                      }
+                      onClick={() =>
+                        addToCartHandler(data, itemsInCart, cartDispatch)
+                      }
+                    >
+                      Add to cart {"   "}
+                      <i className="fa fa-shopping-cart" aria-hidden="true"></i>
+                    </button>
+                    {}
+
+                    {}
+                  </div>
+                </div>
+              );
+            })
+          ) : (
+            <div>
+              <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
+            </div>

```

Comment:
Same for spinner too. In this way, you can reuse components wherever you want your application